internal/castai/castai.go

```diff
@@ -96,7 +102,22 @@ func createHTTPTransport() *http.Transport {
 		IdleConnTimeout:       90 * time.Second,
 		TLSHandshakeTimeout:   5 * time.Second,
 		ExpectContinueTimeout: 1 * time.Second,
+		TLSClientConfig:       tlsConfig,
+	}
+}
+
+func createTLSConfig() (*tls.Config, error) {
+	if cert := config.Get().TLS; cert != nil {
+		certPool := x509.NewCertPool()
+		ok := certPool.AppendCertsFromPEM([]byte(cert.CACertFile))
+		if !ok {
+			return nil, fmt.Errorf("failed to add root certificate to CA pool")
+		}
+		return &tls.Config{
+			RootCAs: certPool,
+		}, nil
 	}
+	return nil, nil
```

Comment:
Maybe add an error message here, if no `cert` is found?

Reply:
`TLSClientConfig` uses default configuration if set to `nil`, so if no optional TLS configuration is found in env vars - `createTLSConfig` just returns default `nil` with no error for the standard agent operation. 
Error is returned in case TLS config is set but certificate provided fails to be parsed as PEM encoded.